File: 2244-minimum-rounds-to-complete-all-tasks/2244-minimum-rounds-to-complete-all-tasks.java
class Solution {

    public int minimumRounds(int[] tasks) {
        int n = tasks.length;
        HashMap<Integer, Integer> map = new HashMap<>();
        for (int ele : tasks) {
            map.put(ele, map.getOrDefault(ele, 0) + 1);
        }
        Set<Integer> set = map.keySet();

        int ans = 0;
        for (int ele : set) {
            int freq = map.get(ele);
         
            if (freq < 3) {
                if (freq % 2 == 1) return -1;
                ans += freq / 2;
            } else {
                if (freq % 3 == 1) {
                     freq-=2;
                      ans += (freq / 3) + (((freq % 3)) / 2)  +1;
                } else {
                    ans += (freq / 3) + ((freq % 3)) / 2;
                }
            }
        }
        return ans;
    }
}

File: 2244-minimum-rounds-to-complete-all-tasks/2244-minimum-rounds-to-complete-all-tasks.py
class Solution(object):
    def minimumRounds(self, tasks):
        dic={}
        cnt=0
        for ele in tasks:
            dic[ele]=dic.get(ele,0)+1
        for val in dic.values():
            if val < 2:
                return -1
            if val%3 == 1:
                val-=2
                cnt+=1
            cnt=cnt+val/3 +(val%3)/2;
        return cnt
        
        